### PR TITLE
Add GET request handler

### PR DESCRIPTION
**Pull request by hakbailey**
#### What does this PR do?
Adds functionality to handle GET requests and respond with challenge secret if provided, or with an appropriate error message if not. Also responds to any non-GET methods with an error message.

#### How can a reviewer manually see the effects of these changes?
Follow the instructions in the README to run and test the Docker container locally. We can't test the function URL locally, so that will have to be verified after merge.

#### Includes new or updated dependencies?
NO

#### What are the relevant tickets?
- https://mitlibraries.atlassian.net/browse/ENSY-66

#### Developer
- [x] All new ENV is documented in README
- [x] Stakeholder approval has been confirmed (or is not needed)

#### Code Reviewer
- [ ] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [ ] There are appropriate tests covering any new functionality
- [ ] The documentation has been updated or is unnecessary
- [ ] The changes have been verified
- [ ] New dependencies are appropriate or there were no changes

### CODE REVIEW

File: lambdas/webhook.py [hunk 1]
```diff
@@ -15,4 +16,34 @@ def lambda_handler(event: dict, context: object) -> str:
...
+            "statusCode": 405,
+            "body": "HTTP method not allowed. Only GET requests are supported.",
+        }
+
+    return base_response | result
+
+
+def handle_get_request(event: dict) -> dict[str, object]:
+    try:
+        challenge_secret = event["queryStringParameters"]["challenge"]
```
Comment (ehanson8) [inline lambdas/webhook.py:40]: Just confirming, there's no validation that happens with the `challenge` parameter? Any value works?

**Comment by hakbailey** [comment]
@cabutlermit can you confirm that these Dockerfile and Makefile changes will allow us to deploy correctly with Github Actions?

File: lambdas/webhook.py (hunk 1 above)
Comment (hakbailey) [reply lambdas/webhook.py:40]: Correct, that gets configured (by us) in Alma, and all our webhook listener needs to do is send back exactly what Alma sent us.

**Comment by cabutlermit** [comment]
> @cabutlermit can you confirm that these Dockerfile and Makefile changes will allow us to deploy correctly with Github Actions?

I can confirm that the `make dist-dev` and `make publish-dev` worked properly from my laptop and published a functional lambda (you can go see it in the AWS Console). But, for GitHub Actions to work, you need an new workflow file (like the `dev_ecr_push.yml` file in the transmogrifier app). That will set up the authentication from GHA to AWS to run the `make` commands. The only difference between the workflow in transmogrifier and the one in this app is the IAM_ROLE environment variable. For this app it should be **"alma-webhook-lambdas-gha-dev"**.

**Comment by hakbailey** [comment]
> > @cabutlermit can you confirm that these Dockerfile and Makefile changes will allow us to deploy correctly with Github Actions?
> 
> I can confirm that the `make dist-dev` and `make publish-dev` worked properly from my laptop and published a functional lambda (you can go see it in the AWS Console). But, for GitHub Actions to work, you need an new workflow file (like the `dev_ecr_push.yml` file in the transmogrifier app). That will set up the authentication from GHA to AWS to run the `make` commands. The only difference between the workflow in transmogrifier and the one in this app is the IAM_ROLE environment variable. For this app it should be **"alma-webhook-lambdas-gha-dev"**.

Cool, thanks! I can do that now or we can wait for Aaron to be back and do it next week, it isn't urgent (I can manually deploy for testing if needed). Do you have a preference?

**Comment by cabutlermit** [comment]
> Cool, thanks! I can do that now or we can wait for Aaron to be back and do it next week, it isn't urgent (I can manually deploy for testing if needed). Do you have a preference?

We can do it today (or tomorrow morning, based on my crazy schedule today). I have the workflow file ready to go - do you want me to just add a commit to this ensy-66 branch with the workflow ready to go?

**Comment by cabutlermit** [comment]
workflow file added. this has **not** been tested 🤞 . The GHA role is published to Param Store by mitlib-tf-workloads-ecr. If the permissions are not correct, we can work that out next.

**Pull request merged**